types/mongodb-queue/index.d.ts

```diff
@@ -13,8 +14,8 @@ declare namespace mongodbQueue {
         constructor(db: Db, name: string, opts?: QueueOptions);
 
         createIndexes(callback: QueueCallback<string>): void;
-        add(payload: string, callback: QueueCallback<string>): void;
-        add(payload: string, opts: QueueOptions, callback: QueueCallback<string>): void;
+        add(payload: Payload, callback: QueueCallback<string>): void;
```

Comment:
Looks like when the payload is an array, the callback returns an array of ids. I think two signatures `        add(payload: Payload, callback: QueueCallback<string>): void;` and `add(payload: Payload[], callback: QueueCallback<string[]>): void;` might be best. And then the Payload type would just be `type Payload = string | Record<string, unknown>;`

Reply:
Yeah right, except type Payload would still need to maintain the Array part for `Payload[]`.

Reply:
Right, makes sense!

Reply:
I just updated the code.